Include/internal/pycore_object.h

```diff
@@ -242,6 +242,17 @@ static inline int _PyType_SUPPORTS_WEAKREFS(PyTypeObject *type) {
     return (type->tp_weaklistoffset > 0);
 }
 
+/// Method to copy objects for specified number of times inside a buffer
+static inline void
+_objects_repeat(PyObject** items, Py_ssize_t copied, Py_ssize_t len_dest)
```

Comment:
Why not re-use `_PyBytes_Repeat`?

Reply:
The reason for not using `_PyBytes_Repeat` is that there are 3 if statements in that function that are not 
used (due to the reference count handling in the list and tuple versions of repeat). In particular `len_dest == 0` is always false, `len_src == 1` is always false and `src != dest` is always false. 

I refactored the `_objects_repeat` to have the multiplication out of the loop (just as `_PyBytes_Repeat`) and did some more profiling:
```
list(1) repeat 2: Mean +- std dev: [patch_objects_repeat_refactor] 62.5 ns +- 0.8 ns -> [patch__PyBytes_Repeat] 69.4 ns +- 12.2 ns: 1.11x slower
list(1) repeat inplace 2: Mean +- std dev: [patch_objects_repeat_refactor] 263 ns +- 2 ns -> [patch__PyBytes_Repeat] 277 ns +- 15 ns: 1.05x slower
list(1) repeat 10: Mean +- std dev: [patch_objects_repeat_refactor] 71.7 ns +- 1.2 ns -> [patch__PyBytes_Repeat] 74.0 ns +- 4.1 ns: 1.03x slower
list(1) repeat inplace 10: Mean +- std dev: [patch_objects_repeat_refactor] 270 ns +- 2 ns -> [patch__PyBytes_Repeat] 276 ns +- 10 ns: 1.02x slower
list(3) repeat 2: Mean +- std dev: [patch_objects_repeat_refactor] 68.4 ns +- 1.7 ns -> [patch__PyBytes_Repeat] 70.5 ns +- 2.8 ns: 1.03x slower
list(3) repeat 10: Mean +- std dev: [patch_objects_repeat_refactor] 116 ns +- 5 ns -> [patch__PyBytes_Repeat] 119 ns +- 2 ns: 1.02x slower
list(3) repeat inplace 10: Mean +- std dev: [patch_objects_repeat_refactor] 305 ns +- 1 ns -> [patch__PyBytes_Repeat] 309 ns +- 4 ns: 1.01x slower
list(10) repeat 2: Mean +- std dev: [patch_objects_repeat_refactor] 98.4 ns +- 3.4 ns -> [patch__PyBytes_Repeat] 99.8 ns +- 3.5 ns: 1.01x slower
list(10) repeat inplace 2: Mean +- std dev: [patch_objects_repeat_refactor] 328 ns +- 3 ns -> [patch__PyBytes_Repeat] 335 ns +- 11 ns: 1.02x slower
list(10) repeat inplace 10: Mean +- std dev: [patch_objects_repeat_refactor] 428 ns +- 8 ns -> [patch__PyBytes_Repeat] 431 ns +- 4 ns: 1.01x slower
list(40) repeat 2: Mean +- std dev: [patch_objects_repeat_refactor] 267 ns +- 2 ns -> [patch__PyBytes_Repeat] 269 ns +- 3 ns: 1.01x slower
list(40) repeat inplace 2: Mean +- std dev: [patch_objects_repeat_refactor] 569 ns +- 6 ns -> [patch__PyBytes_Repeat] 583 ns +- 17 ns: 1.03x slower
list(40) repeat 10: Mean +- std dev: [patch_objects_repeat_refactor] 901 ns +- 3 ns -> [patch__PyBytes_Repeat] 905 ns +- 5 ns: 1.00x slower
list(40) repeat inplace 10: Mean +- std dev: [patch_objects_repeat_refactor] 870 ns +- 21 ns -> [patch__PyBytes_Repeat] 887 ns +- 17 ns: 1.02x slower

Benchmark hidden because not significant (6): list(3) repeat inplace 2, list(10) repeat 10, list(100) repeat 2, list(100) repeat inplace 2, list(100) repeat 10, list(100) repeat inplace 10

Geometric mean: 1.02x slower
```
The version with `_PyBytes_Repeat` is slower, so I would prefer to use the direct call to `_objects_repeat`.

Reply:
@sweeneyde I updated the first comment of this PR with the current status. Your only open comment was on whether to re-use the `_PyBytes_Repeat` method. For performance reasons I would keep the current implementation (see the comment above)